src/CircuitZKit.ts

```diff
@@ -0,0 +1,224 @@
+      ptauFile: "",
+      circuitId,
+      verifierId,
+      wasmBytes: fs.readFileSync(require.resolve("@distributedlab/circom2/circom.wasm")),
+      groth16Template: fs.readFileSync(path.join(__dirname, "templates", "verifier_groth16.sol.ejs"), "utf8"),
+    };
+  }
+
+  public async compile(options = {} as CompileOptions): Promise<void> {
+    createDirIfNotExists(this.config.circuitOutDir);
+
+    const args = this._getCompileArgs({
+      sym: options.sym ?? true,
+      c: options.c ?? false,
+    });
+
+    await this._compile(args, options.quiet ?? true);
+
+    await this._fetchPtauFile();
+    await this._generateZKey();
+    await this._generateVKey();
+  }
+
+  public async generateProof(inputs: I): Promise<ProofStruct> {
+    ensureFileExists(this.config.zKeyFile);
+    ensureFileExists(this.config.wasmFile);
+
+    return (await snarkjs.groth16.fullProve(inputs, this.config.wasmFile, this.config.zKeyFile)) as ProofStruct;
+  }
+
+  public async verifyProof(proof: ProofStruct): Promise<boolean> {
+    ensureFileExists(this.config.vKeyFile);
+
+    const verifier = JSON.parse(fs.readFileSync(this.config.vKeyFile).toString());
+
+    return await snarkjs.groth16.verify(verifier, proof.publicSignals, proof.proof);
+  }
+
+  public async createVerifier(): Promise<void> {
+    createDirIfNotExists(this.config.verifierOutDir);
+    ensureFileExists(this.config.zKeyFile);
+
+    let verifierCode = await snarkjs.zKey.exportSolidityVerifier(this.config.zKeyFile, {
+      groth16: this.config.groth16Template,
+    });
+
+    verifierCode = verifierCode.replace("contract Groth16Verifier", `contract ${this.config.verifierId}`);
+
+    fs.writeFileSync(this.config.verifierFile, verifierCode, "utf-8");
+  }
+
+  public async generateCalldata(proof: ProofStruct): Promise<Calldata> {
+    const calldata = await snarkjs.groth16.exportSolidityCallData(proof.proof, proof.publicSignals);
+
+    return JSON.parse(`[${calldata}]`) as Calldata;
+  }
+
+  private async _generateZKey(): Promise<void> {
+    await snarkjs.zKey.newZKey(this.config.r1csFile, this.config.ptauFile, this.config.zKeyFile);
+  }
+
+  private async _generateVKey(): Promise<void> {
+    const vKeyData = await snarkjs.zKey.exportVerificationKey(this.config.zKeyFile);
+
+    fs.writeFileSync(this.config.vKeyFile, JSON.stringify(vKeyData));
+  }
+
+  private _getCompileArgs(options: CompileOptions): string[] {
+    let args = [this.config.circuitFile, "--r1cs", "--wasm"];
+
+    options.sym && args.push("--sym");
+    options.c && args.push("--c");
+
+    args.push("-o", this.config.circuitOutDir);
+
+    return args;
+  }
+
+  private async _compile(args: string[], quiet: boolean): Promise<void> {
+    try {
+      await this._getCircomRunner(args, quiet).execute(this.config.wasmBytes);
+    } catch {
+      if (quiet) {
+        throw new Error(
+          'Compilation failed with an unknown error. Consider pass "quiet=false" flag to see the compilation error.',
```

Comment:
```suggestion
          'Compilation failed with an unknown error. Consider passing "quiet=false" flag to see the compilation error.',
```

I think it should be `true` by default